### PR TITLE
MAVSDK testing

### PR DESCRIPTION
**Pull request by julianoes**
Work in progress to add integration tests using MAVSDK.

FYI @Stifael, @dagar  

Currently needs MAVSDK installed system-wide (e.g. `/usr/local/`).

Todo: 
- [x] Integrate into PX4 build
- [ ] Add log analysis from @Stifael 
- [ ] Add overall timeout checks.
- [ ] Add scripting around a test matrix, PX4 build and shutdown
- [ ] Add to CI
- [ ] Make sure tests can run in CI, local, as well as in the real world.

To build:
```
PX4_MAVSDK_TESTING=y DONT_RUN=1 make px4_sitl gazebo mavsdk_tests
```

To run:
```
test/mavsdk_tests/mavsdk_test_runner.py --speed-factor 40
```

Closes https://github.com/mavlink/MAVSDK/issues/885.

### CODE REVIEW

File: mavsdk_tests/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+build
```
Comment (dagar) [inline mavsdk_tests/.gitignore]: Can this all live one level down within the existing `test/` folder?

Comment (julianoes) [reply mavsdk_tests/.gitignore]: I guess so, and then it needs to integrate into main build system and all?

**Comment by dagar** [comment]
Why catch2 instead of gtest?

**Comment by julianoes** [comment]
> Why catch2 instead of gtest?

@dagar:
- We already have gtest in other places a d I didn't want to bother with the dependencies. catch2 is header only, so it was easy.
- Now more importantly gtest doesn't use exceptions and therefore a failed assert inside  function (not in the top level test) won't cancel the test but is just a macro to `return false` that you would have to check for. Catch2 uses exceptions so you can assert something in some function and itwill jump out. I wanted to write higher level methods around the MAVSDK APIs, so it was important to have that.

**Comment by MaEtUgR** [comment]
Nice, looking forward to understandable and reproducible tests!
Would it be a bad idea to make `mavsdk_tests/` another submodule? Or at least sideload catch 2? It seems strange to me to include 17k lines catch 2 into the main repository, the code does not belong to this project...

**Comment by julianoes** [comment]
> Or at least sideload catch 2? It

Yes that would be reasonable.

**Comment by LorenzMeier** [comment]
@julianoes Could you rebase this PR? I'd like to contribute to it.

**Pull request closed without merging**